app/components-react/windows/settings/Stream.tsx

```diff
@@ -0,0 +1,435 @@
+      return;
+    }
+
+    this.fixUrl();
+
+    const destinations = cloneDeep(this.customDestinations);
+    const isUpdateMode = typeof this.state.editCustomDestMode === 'number';
+    if (isUpdateMode) {
+      const ind = this.state.editCustomDestMode as number;
+      destinations.splice(ind, 1, this.state.customDestForm);
+    } else {
+      destinations.push(this.state.customDestForm);
+    }
+    this.streamSettingsService.setGoLiveSettings({ customDestinations: destinations });
+    this.stopEditing();
+  }
+
+  private suggestCustomDestName() {
+    const destinations = this.customDestinations;
+    return namingHelpers.suggestName($t('Destination'), (name: string) =>
+      destinations.find(dest => dest.name === name),
+    );
+  }
+}
+
+// wrap the module into a React hook
+function useStreamSettings() {
+  return useModule(StreamSettingsModule).select();
+}
+
+/**
+ * A root component for StreamSettings
+ */
+export function StreamSettings() {
+  const {
+    platforms,
+    protectedModeEnabled,
+    canEditSettings,
+    disableProtectedMode,
+    needToShowWarning,
+    enableProtectedMode,
+  } = useStreamSettings();
+
+  return (
+    <div>
+      {/* account info */}
+      {protectedModeEnabled && (
+        <div>
+          <h2>{$t('Stream Destinations')}</h2>
+          {platforms.map(platform => (
+            <Platform key={platform} platform={platform} />
+          ))}
+
+          <CustomDestinationList />
+
+          {canEditSettings && (
+            <div>
+              <a onClick={disableProtectedMode}>{$t('Stream to custom ingest')}</a>
+            </div>
+          )}
+        </div>
+      )}
+
+      {/* WARNING messages */}
+      {!canEditSettings && (
+        <div className="section section--warning">
+          {$t("You can not change these settings when you're live")}
+        </div>
+      )}
+      {needToShowWarning && (
+        <div className="section section--warning">
+          <b>{$t('Warning')}: </b>
+          {$t(
+            'Streaming to a custom ingest is advanced functionality. Some features may stop working as expected',
+          )}
+          <br />
+          <br />
+
+          {canEditSettings && (
+            <button className="button button--warn" onClick={enableProtectedMode}>
+              {$t('Use recommended settings')}
+            </button>
+          )}
+        </div>
+      )}
+
+      {/* OBS settings */}
+      {!protectedModeEnabled && canEditSettings && <ObsGenericSettingsForm />}
+    </div>
+  );
+}
+
+StreamSettings.page = 'Stream';
+
+/**
+ * Renders a Platform placeholder
+ */
+function Platform(p: { platform: TPlatform }) {
+  const platform = p.platform;
+  const { UserService, StreamingService } = Services;
+  const { canEditSettings, platformMerge, platformUnlink } = useStreamSettings();
+  const isMerged = StreamingService.views.checkPlatformLinked(platform);
+  const username = UserService.state.auth!.platforms[platform]?.username;
```

Comment:
You're accessing `isMerged` and `username` in a non-reactive way, i.e. not using `useVuex`.  Was that intentional?

Reply:
They don't need to be reactive here